plugins/module_utils/redfish_utils.py

```diff
@@ -3394,3 +3394,77 @@ def get_hpe_fan_percent_min(self):
             fan_percent_min_config = hpe.get('FanPercentMinimum')
         result["fan_percent_min"] = fan_percent_min_config
         return result
+
+    def create_volume(self, volume_details):
+        # Find the Storage resource from the requested ComputerSystem resource
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+        storage_uri = data.get('Storage', {}).get('@odata.id')
+        if storage_uri is None:
+            return {'ret': False, 'msg': 'Storage resource not found'}
+
+        # Get Storage Collection
+        response = self.get_request(self.root_uri + storage_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        # Collect Storage Controllers
+        self.storage_controlllers_uris = [i['@odata.id'] for i in response['data'].get('Members', [])]
+        if not self.storage_controlllers_uris:
+            return {
+                'ret': False,
+                'msg': "StorageCollection's Members array is either empty or missing"}
+
+        # Check if storage controller supports volume creation
+        response = self.get_request(self.root_uri + self.storage_controlllers_uris[0])
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        if data.get('Volumes'):
+            response = self.get_request(self.root_uri + data['Volumes']['@odata.id'])
+            if response['ret'] is False:
+                return response
+            data = response['data']
+
+            if data.get('@Redfish.CollectionCapabilities') and data['@Redfish.CollectionCapabilities'].get('Capabilities') \
+                and data['@Redfish.CollectionCapabilities']['Capabilities'][0].get('UseCase') \
+                    and data['@Redfish.CollectionCapabilities']['Capabilities'][0]['UseCase'] == "VolumeCreation":
+                self.storage_controller = self.storage_controlllers_uris[0]
+            else:
+                return {
+                    'ret': False,
+                    'msg': "Storage Controller does not support Volume creation"}
+
+        # Validate input parameters
+        required_parameters = ['RAIDType', 'Links']
```

Comment:
These are not required. In the Volume schema from SNIA, no properties are mandatory for the client to provide. It's expected that the service will fill in default values as needed if the client does not specify certain properties.

Perhaps from an Ansible perspective we'd want to require a user to specify certain thinks in their playbook. It does seem a bit strange for a user to attempt to create a new volume and not specify things like the RAID type or the size.

Reply:
This basic validation of inputs is provided for other modules, e.g. SetBiosAttributes where we seem to be validating if the user provided attributes are valid before doing a POST call. We followed the same here. 
Without ['RAIDType', 'Links'] in the body, it throws the following response on POST.

```
{
    "error": {
        "code": "iLO.0.10.ExtendedInfo",
        "message": "See @Message.ExtendedInfo for more information.",
        "@Message.ExtendedInfo": [
            {
                "MessageId": "iLO.2.15.UnsupportedOperation"
            }
        ]
    }
}

```

Reply:
any updates on this?

Reply:
I've been thinking about this a bit more with others and having a minimum set of inputs does make sense. At a bare minimum, a user should be providing RAIDType, CapacityBytes, and Drives. I think it would be good from an Ansible perspective to abstract the fact that the Redfish model puts "Drives" under links.

Reply:
Changed as suggested, new input format looks like
`Name: "MR Volume"
RAIDType: "RAID0"
Drives:
    - "@odata.id": "/redfish/v1/Systems/1/Storage/DE00B000/Drives/1"`

Reply:
Resolving this conversation. Thanks!